hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java

```diff
@@ -188,6 +188,26 @@ private void associateAccounts(
         for (final var customFee : requireCollectorAutoAssociation) {
             // This should exist as it is validated in validateSemantics
             final var collector = accountStore.get(customFee.feeCollectorAccountIdOrThrow());
+            // TODO: probably this is not the fix but I leave it here to start a discussion.
+            // onlyValidCustomFeeScheduleCanBeCreated was failing because of this:
```

Comment:
I think what you did is correct. But we should use similar logic like mono-service using set of accountIds in a [set](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/token/process/NewRels.java#L36) like this ?